app/javascript/components/URIRef.jsx

```diff
@@ -0,0 +1,64 @@
+import React from "react"
+import PropTypes, { string } from "prop-types"
+
+/**
+ * Input component consisting of a simple textbox.
+ *
+ *  Sample Rails view usage:
+ *
+ * ```
+ * <%= react_component(:URIRef, { paramPrefix: 'example', name: 'title', value: { "@id": "http://example.com/vocab#bar"} }) %>
+ * ```
+ *
+ * When used in a form, this will submit the array `example[title][]`
+ * with a single value, `{value: "http://example.com/vocab#bar"}`.
+ */
+class URIRef extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: props.value,
```

Comment:
To align with https://issues.umd.edu/browse/LIBHYDRA-345, the `state` should be structured as:

```
this.state = {
  uri: props.value['@id']
}
```